pkg/julia/manifest/parse.go

```diff
@@ -0,0 +1,118 @@
+package julia
+
+import (
+	"io"
+	"sort"
+
+	"github.com/BurntSushi/toml"
+	dio "github.com/aquasecurity/go-dep-parser/pkg/io"
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+	"github.com/aquasecurity/go-dep-parser/pkg/utils"
+
+	"golang.org/x/xerrors"
+)
+
+type Manifest struct {
+	JuliaVersion   string                  `toml:"julia_version"`
+	ManifestFormat string                  `toml:"manifest_format"`
+	Dependencies   map[string][]Dependency `toml:"deps"` // e.g. [[deps.Foo]]
+}
+type Dependency struct {
+	Dependencies []string `toml:"deps"` // by name. e.g. deps = ["Foo"]
+	UUID         string   `toml:"uuid"`
+	Version      string   `toml:"version"` // not specified for stdlib packages, which are of the Julia version
+}
+
+type Parser struct{}
+
+func NewParser() types.Parser {
+	return &Parser{}
+}
+
+func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
+	var oldDeps map[string][]Dependency
+	var man Manifest
+	decoder := toml.NewDecoder(r)
+	// Try to read the old Manifest format. If that fails, try the new format.
+	if _, err := decoder.Decode(&oldDeps); err != nil {
+		r.Seek(0, io.SeekStart)
+		if _, err := decoder.Decode(&man); err != nil {
+			return nil, nil, xerrors.Errorf("decode error: %w", err)
+		}
+	}
+
+	// We can't know the Julia version on an old manifest.
+	// All newer manifests include a manifest version and a julia version.
+	if man.ManifestFormat == "" {
+		man = Manifest{
+			JuliaVersion:   "unknown",
+			ManifestFormat: "unknown",
+			Dependencies:   oldDeps,
+		}
+	}
+
+	if _, err := r.Seek(0, io.SeekStart); err != nil {
+		return nil, nil, xerrors.Errorf("seek error: %w", err)
+	}
+
+	// naive parser to get line numbers
+	pkgParser := naivePkgParser{r: r}
+	lineNumIdx := pkgParser.parse()
+
+	var libs []types.Library
+	var deps []types.Dependency
+	for name, manifestDeps := range man.Dependencies {
+		if len(manifestDeps) > 1 {
+			return nil, nil, xerrors.Errorf("multiple entries for dep: %s", name)
+		}
+		manifestDep := manifestDeps[0]
+		version := depVersion(manifestDep, man.JuliaVersion)
+		pkgID := utils.PackageID(manifestDep.UUID, version)
+		lib := types.Library{
+			ID:      pkgID,
```

Comment:
If i understand correctly - we can use `uuid` for `ID` field - https://pkgdocs.julialang.org/v1/toml-files/#Multiple-packages-with-the-same-name

Can you also add test for case when there are 2 dependencies with same names, but different versions and uuid.

Reply:
Done - required a small rewrite